### PR TITLE
Copier update, pass Ruff, update deps

### PR DESCRIPTION
**Pull request by SimonHeybrock**

### CODE REVIEW

File: src/sciline/typing.py [hunk 1]
```diff
@@ -21,19 +11,4 @@
...
-    args = get_args(tp)
-    if len(args) != 2 or type(None) not in args:
-        return None
-    return args[0] if args[1] == type(None) else args[1]  # noqa: E721
-
-
-def get_union(tp: Key) -> Optional[Tuple[Any, ...]]:
-    if get_origin(tp) != Union:
-        return None
-    return get_args(tp)
```
Comment (jl-wynen) [inline src/sciline/typing.py:39]: This seems unrelated to the title of this PR. Why did you remove it?

Comment (SimonHeybrock) [reply src/sciline/typing.py:39]: It turned out to be partially broken code left over from the change of Optional handling.

**Review by jl-wynen — APPROVED** [review]
